File: appExpresionesRegularesIOs_1/appExpresionesRegularesIOs_1/ViewController.m
//
//  ViewController.m
//  appExpresionesRegularesIOs_1
//
//  Created by Roberto Esquivel Troncoso on 09/10/22.
////  By: RETBOT

#import "ViewController.h"

@interface ViewController ()
{
    NSInteger opc;
    NSArray *regex1;
}
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    regex1 = @[@"Correo 1",@"Correo 2"];
    
    _pvRegex.dataSource = self;
    _pvRegex.delegate = self;
    opc = 0;
    _lblOpcion.text = @"(Numeros y Letras).(Numeros y Letras)@(Numeros y Letras).(Numeros y Letras).( 2 o mas Letras) \n Ejemplo: alu.19150519@correo.itlalaguna.edu";
    
    UIColor *color = [UIColor colorWithRed: (175./255) green:(200./255) blue:(100./255) alpha:1.0];
    
    self.view.backgroundColor = color;
    
     //  By: RETBOT
    }

- (NSInteger)numberOfComponentsInPickerView:(nonnull UIPickerView *)pickerView {
    return 1;
}

- (NSInteger)pickerView:(nonnull UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component {
    return regex1.count;
}

- (NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component{
    return [regex1 objectAtIndex:row];
}

- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component{
    _lblOpcion.text = [NSString stringWithFormat:@"Opción: %@", [regex1 objectAtIndex:row]];
    opc = row;
    
    switch (opc) {
        case 0:
            
            _lblOpcion.text = @"(Numeros y Letras).(Numeros y Letras)@(Numeros y Letras).(Numeros y Letras).( 2 o mas Letras) \n Ejemplo: alu.19150519@correo.itlalaguna.edu";
            break;
        case 1:
            _lblOpcion.text = @"(Combinacion de numeros, letras, . , -, _)@(Numeros y Letras).(2 y 4 Letras) \n Ejemplo: roberto_es@correo.com";
            break;
    }
}
//  By: RETBOT
- (IBAction)btnValidar:(UIButton *)sender {
    NSString *expresion;
    NSString *mail = _txtCorreoElectronico.text;
    switch (opc) {
        case 0:
            expresion = @"^[\\w-]+(\\.[\\w-]+)@[A-Za-z0-9]+(\\.[A-Za-z0-9]+)(\\.[A-Z-a-z]{2,})$";
            // inicia con caracteres seguido de un punto
        // despues mas caracteres mas un @
        // continua con alguna palabra esta puede contener, minusculas, mayusculas y numeros seguido de un punto y repitiendo lo mismo que la palabra anterior
        // despues un punto y seguido de 2 o mas caracteres minusculas o mayusculas
                                
            break;
        case 1:
            expresion = @"^[_A-Z0-9a-z.-_]+@[A-Za-z0-9]+\\.[A-Z-a-z]{2,4}$";
            // inicia palabra que contenga letras mayusculas o minusculas, numeros , ".", "_"
            // despues va un @
            // continua con una plabra con minuscula o minusculas y numeros
            // separandolo con un punto
            // la siguiente es una palabra con letras minusculas o mayusculas, de 2 a 4 palabras
            break;
    }
    
    NSError *error = nil;
    NSRegularExpression *regex1 = [NSRegularExpression regularExpressionWithPattern:expresion options:0 error:&error];
    
    NSTextCheckingResult *match = [ regex1 firstMatchInString:mail options:0 range:NSMakeRange(0, [mail length])];
    
    if(match)
        _lblResultado.text = @"Mail valido";
    else
        _lblResultado.text = @"Mail no valido";
}
@end
//  By: RETBOT

File: appPickerSwift1/appPicker1/ViewController.swift
//
//  ViewController.swift
//  appPicker1
//
//  Created by Guest User on 25/11/22.
//
//  By: RETBOT
import UIKit

class ViewController: UIViewController, UIPickerViewDelegate, UIPickerViewDataSource {
    @IBOutlet weak var txtCarro1: UITextField!
    @IBOutlet weak var pickerViewCarros: UIPickerView!
    
    let carroColor = [["VW", "Ford", "Toyota", "Nissan", "Toyota", "Triciclos Apache", "Peugot"], ["Azul🐬", "Verde🥑", "Amarillo🍋", "Rojo🐙", "Negro🖤"]]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        //let pickerView1 = UIPickerView()
        pickerViewCarros.delegate = self
        pickerViewCarros.dataSource = self
        //txtCarro1.inputView = pickerViewCarros
    }
    //  By: RETBOT
    func numberOfComponents(in pickerView: UIPickerView) -> Int {
        return 2;
    }
    
    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
        return carroColor[component].count
    }
    
    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
        return self.carroColor[component][row]
    }
    
    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
        let carro = carroColor[0][pickerView.selectedRow(inComponent: 0)]
        let color = carroColor[1][pickerView.selectedRow(inComponent: 1)]
        txtCarro1.text = carro + " " + color;
    }


}//  By: RETBOT

